Slim/Player/TranscodingHelper.pm

```diff
@@ -89,6 +89,11 @@ sub loadConversionTables {
 				my $clientid   = lc($4);
 				my $profile = "$inputtype-$outputtype-$clienttype-$clientid";
 
+				if ($commandTable{$profile}) {
```

Comment:
This block has taken me at least 10 minutes to make me believe I understand what it's supposed to do. It should add an index number to the profile? But I still don't see it do anything. Is this only being executed under certain circumstances we don't meet in a default installation?

While the implementation might be correct, it's almost too perly for me :-). Could you please split up that statement into human digestible chunks?

Reply:
It took me more than 10 minutes to build it ;-). The idea is to add a number N to the profile when there is more than one "$inputtype-$outputtype-$clienttype-$clientid"
- if there is a rule with the "base" matching, means we have to stitch a number
- find all the profiles with that match the rule
- sort them, 
- because they are all identical except the number at the end, the last sorted item is the highest number
- extract the number, add one and stitch that to the profile

With that, when there is only one profile, there is no change to the naming and as soon as there is more than one, it has a added tail of '-x' (where x 1...). 
I know it's Perly, but I'm also trying to use the language for what it does best, being compact and offering this great usage of regex, arrays and hashes